lang/src/org/partiql/lang/eval/like/Pattern.kt

```diff
@@ -0,0 +1,89 @@
+package org.partiql.lang.eval.like
+
+import java.util.regex.Pattern
+
+private const val ANY_MANY = '%'.toInt()
+private const val ANY_ONE = '_'.toInt()
+
+/**
+ * Translates a SQL-style `LIKE` pattern to a regular expression.
+ *
+ * @param likePattern A `LIKE` match pattern (i.e. a string where '%' means '.*?' and '_' means '.').
+ * @param escapeChar The escape character for the `LIKE` pattern.
+ *
+ * @return a [Pattern] which is a regular expression corresponding to the specified `LIKE` pattern.
```

Comment:
Perhaps add an example of a `LIKE` pattern and the equivalent regex?

Reply:
There are some examples in PatternTests.kt but some equivalent regex would be nice here